core/src/test/java/org/elasticsearch/search/aggregations/metrics/percentiles/hdr/InternalHDRPercentilesRanksTests.java

```diff
@@ -61,4 +66,26 @@ protected void assertReduced(InternalHDRPercentileRanks reduced, List<InternalHD
         return InternalHDRPercentileRanks::new;
     }
 
+    @Override
+    protected NamedXContentRegistry xContentRegistry() {
+        return new NamedXContentRegistry(singletonList(
+                new NamedXContentRegistry.Entry(
+                        Aggregation.class,
+                        new ParseField(InternalHDRPercentileRanks.NAME),
+                        (parser, context) -> ParsedHDRPercentileRanks.fromXContent(parser, (String) context))));
```

Comment:
I am ok with a followup as well. but in the previous comments you said "when we have many parsable aggregations". I would do it earlier than that :)